Assets/Scripts/Enemies/BaseEnemyAI.cs

```diff
@@ -0,0 +1,136 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BaseEnemyAI : MonoBehaviour
+{
+    public int currentHealth;
+
+    [SerializeField] Animator animator;
+    [SerializeField] MovementOverride movement;
+    [SerializeField] Rigidbody2D rb;
+    [SerializeField] Resource health;
+    [SerializeField] AttackData meleeAttackData;
+    [SerializeField] AttackData projectileAttackData;
+
+    [SerializeField] float detectionRadius;
+    [SerializeField] float meleeRange;
+    [SerializeField] Transform projectileSpawnPoint;
+
+    PlayerMovementController player;
+    PlayerCombat playerCombat;
+
+    Vector2 velocity;
+    bool isFacingRight = true;
+    float horizontalMove = 0.0f;
+    float maxSpeedChange;
+    float nextMeleeTime = 0f;
+    float nextProjectileTime = 0f;
+
+    // Values used for animations:
+
+    [SerializeField] bool isFiringProjectile = false;
+    [SerializeField] bool isAttacking = false;
+
+    void FindPlayer()
+    {
+        if (player == null)
+        {
+            player = GameObject.FindWithTag(Constants.PLAYER_TAG).GetComponent<PlayerMovementController>();
```

Comment:
If the player object is not found, this will throw an error. Recommend:

```suggestion
            GameObject playerObj = GameObject.FindWithTag(Constants.PLAYER_TAG);
            if (playerObj != null) player = playerObj.GetComponent<PlayerMovementController>();
```

Reply:
When the player dies, we will likely want to wait for some duration of time before respawning, so there will likely be cases where the player object is not present in the scene

Reply:
I believe this is addressed with the new coroutine